ambari-server/src/main/java/org/apache/ambari/server/upgrade/UpgradeCatalog300.java

```diff
@@ -561,6 +564,32 @@ protected void executeDMLUpdates() throws AmbariException, SQLException {
     updateKerberosConfigurations();
     upgradeLdapConfiguration();
     createRoleAuthorizations();
+    addUserAuthenticationSequence();
+  }
+
+  protected void addUserAuthenticationSequence() throws SQLException {
+    final long maxUserAuthenticationId = fetchMaxUserAuthenticationId();
+    LOG.info("Maximum user authentication ID = " + maxUserAuthenticationId);
+    addSequence("user_authentication_id_seq", maxUserAuthenticationId + 1, false);
+  }
+
+  private long fetchMaxUserAuthenticationId() throws SQLException {
+    Statement statement = null;
+    ResultSet rs = null;
+    try {
```

Comment:
Can be simplified a bit using "try-with":

```
    try (Statement stmt = dbAccessor.getConnection().createStatement();
         ResultSet rs = stmt.executeQuery(String.format("SELECT MAX(user_authentication_id) FROM %s", USER_AUTHENTICATION_TABLE))
    ) {
      if (rs.next()) {
        return rs.getLong(1);
      }
    }
```

Reply:
Good point